app/src/androidTest/java/at/tugraz/ist/guessingwords/data/service/WordServiceTest.kt

```diff
@@ -128,4 +130,109 @@ class WordServiceTest {
         assert(cbGet.retWord!!.text == word.text)
         assert(cbGet.retWord!!.uid == cb.uid)
     }
+    @Test(timeout = 3000)
+    fun checkIfNewMultiplayerWordPoolWorksAndCheckForWordsEntered() {
+
+        val word = Word("test1")
+        val cb = object:Callback<Long>{
+            val finished = ConditionVariable()
+            override fun whenReady(data: Long?) {
+                finished.open()
+            }
+        }
+        service.insertOrUpdateExistingWord(word, cb)
+        cb.finished.block()
+
+        // In memory Instance create
+        val cbInMemory = object:Callback<List<Long>>{
+            val finished = ConditionVariable()
+            var resWordIds = listOf<Long>()
+            override fun whenReady(data: List<Long>?) {
+                resWordIds = data!!
+                finished.open()
+            }
+        }
+        service.createNewMultiplayerWordPool(cbInMemory)
+        cbInMemory.finished.block()
+        //Merge Some words
+
+        val cbMerge = object:Callback<List<Long>>{
+            val finished = ConditionVariable()
+            var resWordIds = listOf<Long>()
+            override fun whenReady(data: List<Long>?) {
+                resWordIds = data!!
+                finished.open()
+            }
+        }
+        val wordList = listOf(Word("test1"), Word("test2"), Word("test3"))
+        service.mergeIntoDatabase(wordList, cbMerge)
+        cbMerge.finished.block()
+
+        // get all Words test
+        val cbGetAll = object:Callback<List<Word>>{
+            val finished = ConditionVariable()
+            var resWords = listOf<Word>()
+            override fun whenReady(data: List<Word>?) {
+                resWords = data!!
+                finished.open()
+            }
+        }
+        service.getAllWords(cbGetAll)
+        cbGetAll.finished.block()
+        assert(cbGetAll.resWords.count() >= 3)
+        service.removeMultiplayerWordPool()
+    }
+
+    @OptIn(ExperimentalStdlibApi::class)
+    @Test(timeout = 3000)
+    fun checkForDuplicatesAfterMerge(){
+        val word = Word("test1")
+        val cb = object:Callback<Long>{
+            val finished = ConditionVariable()
+            override fun whenReady(data: Long?) {
+                finished.open()
+            }
+        }
+        service.insertOrUpdateExistingWord(word, cb)
+        cb.finished.block()
+
+        // In memory Instance create
+        val cbInMemory = object:Callback<List<Long>>{
+            val finished = ConditionVariable()
+            override fun whenReady(data: List<Long>?) {
+                finished.open()
+            }
+        }
+        service.createNewMultiplayerWordPool(cbInMemory)
+        cbInMemory.finished.block()
```

Comment:
You make the dangerous assumption that "test1" will actually already be in the database.
That is never checked.
Write a new test to actually assert that.